chroma-manager/chroma_core/plugins/linux.py

```diff
@@ -446,61 +454,82 @@ def preferred_serial(bdev):
             parent_resource = self.major_minor_to_node_resource[bdev['parent']]
 
             if not parent_resource.logical_drive:
-                raise RuntimeError("Parent %s of %s has no logical drive" % (parent_resource, bdev))
-
-            partition, created = self.update_or_create(Partition,
-                    parents = [parent_resource],
-                    container = parent_resource.logical_drive,
-                    number = bdev['partition_number'],
-                    size = bdev['size'],
-                    filesystem_type = bdev['filesystem_type'])
+                raise RuntimeError("Parent %s of %s has no logical drive" %
+                                   (parent_resource, bdev))
+
+            partition, created = self.update_or_create(
+                # ZfsPartitions should be differentiated as they are not usable for lustre
+                ZfsPartition
```

Comment:
It might read a little cleaner if these three lines were in a function:
```
partition, created = self.update_or_create(
   getPartition(bdev),
   parnets=[parent_resource]
   ...)
```